src/core/execute.c

```diff
@@ -7042,6 +7045,8 @@ void exec_params_clear(ExecParameters *p) {
 
         p->environment = strv_free(p->environment);
         p->fd_names = strv_free(p->fd_names);
+        for (size_t i = p->n_socket_fds + p->n_storage_fds; i < p->n_socket_fds + p->n_storage_fds + p->n_openfile_fds; ++i)
+                safe_close(p->fds[i]);
```

Comment:
hmm, is this right? so far we never considered the fds to become part of ExecParameters

Reply:
`close_many` won't work here since we don't start closing `fds` at index 0.

Is it possible to leverage the existing LISTEN_FDS mechanism without putting the OpenFile fds in the ExecParameters?

Reply:
```c
close_many(p->fds + p->n_socket_fds + p->n_storage_fds, p->n_openfile_fds);
```